Magento2/Helpers/Commenting/PHPDocFormattingValidator.php

```diff
@@ -122,13 +122,18 @@ public function hasDeprecatedWellFormatted($commentStartPtr, $tokens)
             return true;
         }
 
+        $seeTagRequired = false;
         if ($tokens[$deprecatedPtr + 2]['code'] !== T_DOC_COMMENT_STRING) {
-            return false;
+            $seeTagRequired = true;
         }
 
         $seePtr = $this->getTagPosition('@see', $commentStartPtr, $tokens);
         if ($seePtr === -1) {
-            return true;
+            if ($seeTagRequired) {
+                return false;
+            } else {
```

Comment:
`else` can be eliminated here.

Reply:
If we remove "else" branch there, it could lead to unexpected behavior because of the next IF condition:
`
if ($tokens[$seePtr + 2]['code'] !== T_DOC_COMMENT_STRING) {
            return false;
        }
`
That happens because $seePtr could be -1, so $seePtr + 2 points to how-knows-what.
"Else" part is required because it covers behavior when @see is not present at all.

Reply:
Actually, it really fails on some regular tests when I remove "else" branch.

Reply:
`$seeTagRequired` is boolean and if it is `false`, execution will drop to `else` or directly to the `return`.

I propose to simplify code even more:
```
        $seePtr = $this->getTagPosition('@see', $commentStartPtr, $tokens);
        if ($seePtr === -1) {
            return !$seeTagRequired;
        }
        return $tokens[$seePtr + 2]['code'] === T_DOC_COMMENT_STRING;
```